site/packages/aa-core/smart-account-client/middleware/index.md

```diff
@@ -0,0 +1,79 @@
+---
+outline: deep
+head:
+  - - meta
+    - property: og:title
+      content: SmartContractAccount
+  - - meta
+    - name: description
+      content: Overview of ClientMiddleware exported by aa-core
+  - - meta
+    - property: og:description
+      content: Overview of ClientMiddleware exported by aa-core
+---
+
+# ClientMiddleware
+
+Middleware represents different operations involved in the [`SmartAccountClient`](/packages/aa-core/smart-account-client/) pipeline for constructing a user operation given the user inputs by populating the UO with other data including gas fee, paymaster data, etc.
+
+Each middleware is a function that takes in a user operation object, performs its job to retrieve or compute the data, and populate different fields of user operation to pass onto the next middleware in the pipeline before being signed and sent to the network.
+
+```ts
+export type ClientMiddlewareFn = <TAccount extends SmartContractAccount>(
+  struct: Deferrable<UserOperationStruct>,
+  args: {
+    overrides?: UserOperationOverrides;
+    feeOptions?: UserOperationFeeOptions;
+    account: TAccount;
+  }
+) => Promise<Deferrable<UserOperationStruct>>;
+
+export type ClientMiddleware = {
```

Comment:
this is missing: `dummyPaymasterAndData: ClientMiddlewareFn;`

Reply:
I intentionally removed it because that's not something that users can or should override